src/zac/core/camunda.py

```
@@ -62,8 +62,30 @@ def get_task(task_id: CamundaId) -> Optional[Task]:
         data = client.get(f"task/{task_id}")
     except requests.HTTPError as exc:
         if exc.response.status_code == 404:
-            return None
-        raise
+            # see if we can get it from the history
+            historical = client.get("history/task", {"taskId": task_id})
+            if not historical:
+                return None
+
+            assert (
+                len(historical) < 2
+            ), f"Found multiple tasks in the history for ID {task_id}"
```

I don't understand why this should raise an assertion error.

---

The API is weird for me, I don't understand how there's not a detail endpoint for it. The task IDs are UUIDs, so they should be unique. Since we're doing the lookup based on ID, we really expect a single result to be returned - especially since the lookup should be initiated by something that originated from a non-history task. If you're just guessing IDs, then this will 404 for the client.

It's basically a sanity check expressing that we never expect more than one result, and if we do, all systems should yell very loudly to figure out what assumption is wrong.